core/src/main/java/edu/wpi/grip/core/operations/composite/WatershedOperation.java

```diff
@@ -64,12 +65,24 @@
   private final InputSocket<ContoursReport> contoursSocket;
   private final OutputSocket<ContoursReport> outputSocket;
 
+  private static final int MAX_MARKERS = 253;
+  private final List<Mat> markerPool;
+  private final MatVector contour = new MatVector(); // vector with a single element
+  private final Mat markers = new Mat();
+  private final Mat output = new Mat();
+  private final Point backgroundLabel = new Point();
+
   @SuppressWarnings("JavadocMethod")
   public WatershedOperation(InputSocket.Factory inputSocketFactory,
                             OutputSocket.Factory outputSocketFactory) {
     srcSocket = inputSocketFactory.create(srcHint);
     contoursSocket = inputSocketFactory.create(contoursHint);
     outputSocket = outputSocketFactory.create(outputHint);
+    List<Mat> tmpPool = new ArrayList<>();
+    for (int i = 0; i < MAX_MARKERS; i++) {
+      tmpPool.add(new Mat());
+    }
+    markerPool = ImmutableList.copyOf(tmpPool);
```

Comment:
If we use Guava 21.0:
`markerPool = Stream.generate(Mat::new).limit(MAX_MARKERS).collect(ImmutableList.toImmutableList());`